### ISSUE TITLE
Fix false positives for universal selector and disallowInList in no-duplicate-selectors

### ISSUE THREAD

**Issue by manovotny** [stylelint/stylelint#4683]
> Clearly describe the bug

There are false positives for no-duplicate-selectors's `disallowInList: false`.

In summary, it's flagging some, but not all, selectors after a `*` declaration as a duplicate tag.

I've also created a [demo](https://stylelint.io/demo#N4Igxg9gJgpiBcICWBbADhATgFwATDExgENsYBxAGwgCNjKBlbAT0pgF9cAzTCFXAOQBnFmygBaSOggA7GDOxCBAbgA6M9ZBkjcVWvSasYuALy5CJMnrqNRMAAbrcz3AAtsKSvicvfNCAAe4kJIAF5IMgDm8Lj+mLCY4v4BajK+vj7puABUADSZ6dnw8DQwXFgw+WlZOcXEXGSY3tU1LsnBYRHRuBGuMJhI2Kmt7AWjGi24xM2tkNSYMQDuroMwwy7jGZNoMzVzWDFEUOvO4-ap6jABGDi4sFzEAK6UeNYGdqkguSBCzArEAQQ4CEQnEEXEACshF9wLIuEhIkDgD5VCBMM8YEJUTFkZNUTIIOIoI80JQkGBSDBgjA2GBsFgsQhcABtAq+bDoypsly41oZEBQJBCejURYASRkABkhdhsbgOY8YNzTgUALo+cbsEDsIA) of the issue.

> Which rule, if any, is the bug related to?

`no-duplicate-selectors`

> What code is needed to reproduce the bug?

```js
import {createGlobalStyle} from 'styled-components';

const GlobalStyle = createGlobalStyle`
    html {
        box-sizing: border-box;
        
        *,
        *::before,
        *::after {
            box-sizing: inherit;
        }
    }

    a {
        color: white;
    }
    
    p {
        color: red;
    }
`;

export default GlobalStyle;
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "no-duplicate-selectors": [
        true,
        {
            "disallowInList": true
        }
    ]
  }
}
```

> Which version of stylelint are you using?

13.2.1

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint "**/*.js" --report-needless-disables"`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Could just be a CSS-in-JS issue. Unknown.

> What did you expect to happen?

No errors / warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

The following errors / warnings were flagged:

```shell
14:5 error Unexpected duplicate selector "a", first used at line 7 (no-duplicate-selectors)
```

**Comment by jeddy3** [comment]
@manovotny Thanks for another clear report.

I can [reproduce is with](https://stylelint.io/demo#N4Igxg9gJgpiBcIBU94A8AEwC+AdAdpjiADQgDOAnvgC4CGaC455p4E+AZgJYDmTwAhgy4QAJwCuAGxjlR8LEOEiQ+CAFooEgA5TuYOjRjryMGWBoQxchBgDaS5cpqSYJR08X5PPlVG7kdFJSEADuAJL4ADIBNPIYLhIwHk543soAukpp2CDYQA):

```css
*::x {}
x {}
```

And:

```json
{
  "rules": {
    "no-duplicate-selectors": [
        true,
        {
          "disallowInList": true
        }
    ]
  }
}
```

I believe any matching letter proceeded by the universal selector will trigger the bug. I can't reproduce without the `"disallowInList": true` secondary option. The problem, as you said, lies there.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.
